nomos-da/kzgrs-backend/src/common/mod.rs

```diff
@@ -175,3 +176,56 @@ pub fn commitment_to_bytes(commitment: &Commitment) -> Vec<u8> {
         .expect("Serialization of commitment should work");
     buff.into_inner()
 }
+
+pub fn serialize_canonical<S, T>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+    T: CanonicalSerialize,
+{
+    let mut bytes = Vec::new();
+    value
+        .serialize_compressed(&mut bytes)
+        .map_err(serde::ser::Error::custom)?;
+    serializer.serialize_bytes(&bytes)
+}
+
+pub fn deserialize_canonical<'de, D, T>(deserializer: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: CanonicalDeserialize,
+{
+    let bytes: Vec<u8> = serde::Deserialize::deserialize(deserializer)?;
+    let mut cursor = Cursor::new(bytes);
+    T::deserialize_compressed(&mut cursor).map_err(serde::de::Error::custom)
+}
+
+pub fn serialize_vec_canonical<S, T>(values: &[T], serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+    T: CanonicalSerialize,
+{
+    let mut container = serializer.serialize_seq(Some(values.len()))?;
+    for value in values {
+        let mut bytes = Vec::new();
+        value
+            .serialize_compressed(&mut bytes)
+            .map_err(serde::ser::Error::custom)?;
+        container.serialize_element(&bytes)?;
+    }
+    container.end()
+}
+
+pub fn deserialize_vec_canonical<'de, D, T>(deserializer: D) -> Result<Vec<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: CanonicalDeserialize,
+{
+    let bytes_vecs: Vec<Vec<u8>> = Deserialize::deserialize(deserializer)?;
+    bytes_vecs
+        .iter()
+        .map(|bytes| {
+            let mut cursor = Cursor::new(bytes);
+            T::deserialize_compressed(&mut cursor).map_err(serde::de::Error::custom)
+        })
+        .collect()
+}
```

Comment:
We need to check on this. There must be a better way of deriving this (probably deriving for third party crates. ). Also I think we already had this problem somewhere else. Could you please add an issue pointing to it?

Reply:
I believe you're referring to this https://github.com/logos-co/nomos-node/pull/650#discussion_r1610021174

In this PR I've reused the same serde for Blob and Cert by moving related functions to common module. `ark_serialize` module provides `CanonicalDeserialize` trait implementations for proofs and commitments that then needs a wrapper to work with serde. Maybe you have an example of how you'd like this to be implemented?

Reply:
> Maybe you have an example of how you'd like this to be implemented?

I'll give it a try after this gets merged.